hs_core/hydroshare/utils.py

```diff
@@ -249,20 +258,9 @@ def get_file_from_irods(res_file):
     :return: location of the copied file
     """
     res = res_file.resource
-    if res_file.fed_resource_file or res_file.fed_resource_file_name_or_path:
-        istorage = IrodsStorage('federated')
-    else:
-        istorage = IrodsStorage()
-    if res_file.resource_file:
-        res_file_path = res_file.resource_file.name
-        file_name = os.path.basename(res_file.resource_file.name)
-    elif res_file.fed_resource_file:
-        res_file_path = res_file.fed_resource_file.name
-        file_name = os.path.basename(res_file.fed_resource_file.name)
-    else:
-        res_file_path = os.path.join(res.resource_federation_path, res.short_id,
-                                     res_file.fed_resource_file_name_or_path)
-        file_name = os.path.basename(res_file.fed_resource_file_name_or_path)
+    istorage = res.get_irods_storage()
+    res_file_path = res_file.storage_path
+    file_name = os.path.basename(res_file_path)
```

Comment:
@alvacouch @pkdash I am afraid there might be a chicken-and-egg issue here (although I am not sure since this might not be an issue any more with the changes in this PR). Specifically, if a specific resource type is listening to pre_create_resource signal before the resource is created to extract metadata, the storage_path is most likely not set at this point since the resource is not created yet, but set on ```res_file.fed_resource_file_name_or_path```? Worth checking to make sure.